semgrep/semgrep/config_resolver.py

```diff
@@ -66,42 +70,73 @@
 }
 
 
+class ConfigType(Enum):
+    REGISTRY = 1
+    CDN = 2
+    LOCAL = 3
+
+
 class ConfigPath:
-    _from_registry = False
+    _origin = ConfigType.LOCAL
     _config_path = ""
+    _project_url = None
+    _extra_headers: Dict[str, str] = {}
 
-    def __init__(self, config_str: str) -> None:
+    def __init__(self, config_str: str, project_url: Optional[str]) -> None:
         """
         Mutates MetricManager state!
         Takes a user's inputted config_str and transforms it into the appropriate
         path, checking whether the config string is a registry url or not. If it
         is, also set the appropriate MetricManager flag
         """
+        self._project_url = project_url
+
```

Comment:
```suggestion
        self._project_url = project_url
        self._origin = ConfigType.REGISTRY
```

then only override the two places we differ